### PR TITLE
Feature Controller

### PR DESCRIPTION
**Pull request by Obimbo07**
In this pull request the following is tested:

- If response status was correct.
- if a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -2,6 +2,33 @@
...
+      expect(response).to have_http_status(:success)
+    end
+  end
+  describe 'Elements for posts' do
+    it 'render the template of post index' do
+      user = User.create(name: 'Austin', id: '1')
+      get user_posts_path(user)
+      expect(response.body).to render_template('index')
+      expect(response.body).to include('<h1>Show all users</h1>')
+    end
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb:25]: - [ ] Kindly organize your specs. This it-block is for the index posts page and so should be under the `GET /index` describe block above.

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -2,6 +2,32 @@
...
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'Elements for users' do
+    it 'render the template of post index' do
+      get '/users'
+      expect(response.body).to render_template('index')
+      expect(response.body).to include('<h1>Show all users</h1>')
+    end
```
Comment (julie-ify) [inline spec/requests/users_spec.rb:24]: - [ ] Kindly organize your specs. This it-block is for the index posts page and so should be under the `GET /index` describe block above.

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -2,6 +2,32 @@
...
+
+    it 'render the show template for post' do
```
Comment (julie-ify) [inline spec/requests/users_spec.rb:26]: - [ ] Kindly fix your failing specs. You need to install the `rails-controller-testing` gem as indicated by the error message.

  image

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -2,6 +2,33 @@
 
 RSpec.describe 'Posts', type: :request do
   describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+    it 'returns https success' do
+      user = User.create(name: 'Austin', id: '1')
+      get user_post_path(user)
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb]: - [ ] After adding the gem to Gemfile and running the bundler, this test still fails. You need to specify the correct route: `user_posts_path(user)`

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Obimbo07,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @Obimbo07,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Descriptive PR⭐
- [ ]  No Linters error 🤩
- [ ] Good job working on the testsI💪🏻
- [ ] All tests passed🔥🔥


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**